src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -876,6 +876,29 @@ begin
   ... ≤ R + δ : add_le_add (hR zE) hz.le
 end
 
+lemma exists_disjoint_thickenings {s t : set X} (hs : is_compact s) (ht : is_closed t)
+  (hst : disjoint s t) :
+  ∃ δ, 0 < δ ∧ disjoint (thickening δ s) (thickening δ t) :=
+begin
+  obtain ⟨u, -, u_pos, u_lim⟩ := exists_seq_strict_anti_tendsto (0 : ℝ),
+  suffices h : ∃ (n : ℕ), disjoint (thickening (u n) s) (thickening (u n) t),
+  { obtain ⟨n, hn⟩ := h,
+    exact ⟨u n, u_pos n, hn⟩ },
+  by_contra' h,
+  -- have := hs.exists_forall_le,
```

Comment:
@hrmacbeth, I didn't find the proof using `hs.exists_forall_le`. Am I missing a golf?

Reply:
I don't know where you discussed this with Heather, but the proof that Heather had in mind is very clear, it's the natural proof once you've done enough of this kind of things. However there are a number of related issues here. First you assume pseudo_metric spaces instead of metric, hence you have `edist` everywhere and the infimum distance could be infinite. On top of that you very quickly get into coercion hell between `real`, `nnreal`, `ennreal`. Below is the proof that Heather probably had in mind, modulo coercion hell and a bit of API about `inf_edist` and `thickening`.
```lean
  let d : X → ℝ≥0∞ := λ x, inf_edist x t,
  rcases s.eq_empty_or_nonempty with rfl | Hs,
  { exact ⟨1, zero_lt_one, by simp⟩ },
  { obtain ⟨x, x_in, hx : ∀ y ∈ s, d x ≤ d y⟩ :=
      hs.exists_forall_le Hs continuous_inf_edist.continuous_on,
    by_cases Hx : inf_edist x t = ∞,
    { have key : ∀ y ∈ s, d y = ∞, from λ y y_in,  eq_top_iff.mpr (Hx ▸ hx y y_in),
      refine ⟨1, zero_lt_one, _⟩,
      sorry },
    { lift ((d x)/2) to ℝ≥0 with δ,
      refine ⟨δ, _, _⟩,
      { have : d x ≠ 0,
        { intro Hx,          
          exact hst ⟨x_in, (mem_iff_inf_edist_zero_of_closed ht).mpr Hx⟩ },
        sorry },
      { rintros z ⟨hzs, hzt⟩,
        rcases (mem_thickening_iff _ _).mp hzs with ⟨z₁, z₁_in, hz₁⟩,
        rcases (mem_thickening_iff _ _).mp hzt with ⟨z₂, z₂_in, hz₂⟩,
        apply lt_irrefl (d x),
        
        calc d x ≤ d z₁ : hx z₁ z₁_in
        ... ≤ edist z₁ z₂ : inf_edist_le_edist_of_mem z₂_in
        ... ≤ edist z₁ z + edist z z₂ : edist_triangle z₁ z z₂
        ... < δ + δ : by { rw [edist_comm], sorry }
        ... = d x : by rw [h, (d x).add_halves] },
      { refine λ h, Hx _,
        sorry } } },
```

Reply:
Do you think that's worth changing? Your draft is already longer than my proof.

Reply:
I think it's worth making sure that we have enough lemmas to make that proof nice and short. This will require useful lemmas about `edist`, `inf_edist`, `thickening` and nice lemmas about `ennreal`, `nnreal` and `real`. Probably some of those lemmas already exist. This way you will have both a readable natural proof of this lemma and many useful lemmas.